central/views/imagecve/view_impl.go

```diff
@@ -190,13 +191,46 @@ func (v *imageCVECoreViewImpl) GetImageIDs(ctx context.Context, q *v1.Query) ([]
 }
 
 func withSelectCVEIdentifiersQuery(q *v1.Query) *v1.Query {
+	// For pagination and sort to work properly, the filter query to get the CVEs needs to
+	// include the fields we are sorting on.  At this time custom code is required when
+	// sorting on custom sort fields.  For instance counts on the Severity column based on
+	// a value of that column
+	// TODO(ROX-26310): Update the search framework to inject required select.
+	addSeverities := false
+	severitySort := []string{
+		search.CriticalSeverityCount.String(),
+		search.FixableCriticalSeverityCount.String(),
+		search.ImportantSeverityCount.String(),
+		search.FixableImportantSeverityCount.String(),
+		search.ModerateSeverityCount.String(),
+		search.FixableModerateSeverityCount.String(),
+		search.LowSeverityCount.String(),
+		search.FixableLowSeverityCount.String(),
+	}
 	cloned := q.CloneVT()
 	cloned.Selects = []*v1.QuerySelect{
 		search.NewQuerySelect(search.CVEID).Distinct().Proto(),
 	}
 	cloned.GroupBy = &v1.QueryGroupBy{
 		Fields: []string{search.CVE.String()},
 	}
+
+	// Check to see if a severity sort is in the query
+	for _, severity := range severitySort {
```

Comment:
I think you could use `slices.Contains`, I've run a simple test. This would save the loop and reduce it to a simple if-case:

```
func TestIt(t *testing.T) {
	query := []string{"test", "sort", "more", "severity"}
	assert.True(t, slices.Contains[[]string, string](query, "severity"))
	assert.False(t, slices.Contains[[]string, string](query, "cve"))
}
```

Reply:
nice.  One of these days I will remember some of these slice helpers.

Reply:
Actually not that simple.  I would have to go down and find all the sort options to narrow that down.  I'm looking at the entire pagination string that could in theory contain multiple sort options to see if one of the ones we are handling special exists.  I will probably give it some thought over the weekend.